### DISCUSSION TITLE
Get the instance of the actual model

### DISCUSSION THREAD

**Discussion by josegus** [filamentphp/filament#315] [Help]
It is possible to get the actual model when edit the record?

I'm building a nested categories form, I need to disable the current model, so the user won't select it.

Thought using `self::getModel()` would get the current model

```php
Components\BelongsToSelect::make('category_id')
    ->label('Parent category')
    ->placeholder('--Parent Category--')
    ->relationship('category', 'name', function ($query) {
        return $query->where('id', '!=', self::getModel()->id);
    })
    ->nullable()
    ->preload(),
```

**Comment by ryangjchandler** [comment]
> It is possible to get the actual model when edit the record?
> 
> I'm building a nested categories form, I need to disable the current model, so the user won't select it.
> 
> Thought using `self::getModel()` would get the current model
> 
> ```
> Components\BelongsToSelect::make('category_id')
>     ->label('Parent category')
>     ->placeholder('--Parent Category--')
>     ->relationship('category', 'name', function ($query) {
>         return $query->where('id', '!=', self::getModel()->id);
>     })
>     ->nullable()
>     ->preload(),
> ```

You can use `when()`. That will provide the `$record` which will be the model.

**Reply by ryangjchandler** [reply]
I should provide an example:

```php
->when(fn () => true, function ($field, $record) {

})
```
